executor/distsql.go

```diff
@@ -824,14 +827,25 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
 		sort.Sort(task)
 	}
 
-	if w.isCheckOp && handleCnt != len(task.rows) {
-		obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
-		for _, row := range task.rows {
-			handle := row.GetInt64(w.handleIdx)
-			obtainedHandlesMap[handle] = struct{}{}
+	if handleCnt != len(task.rows) {
+		if w.isCheckOp {
+			obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
+			for _, row := range task.rows {
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d, missing handles %v in a batch",
+				w.idxLookup.index.Name.O, handleCnt, len(task.rows), GetLackHandles(task.handles, obtainedHandlesMap))
+		}
+
+		if len(w.idxLookup.tblPlans) == 1 {
+			// if this table scan has no condition, the number of rows it returns must equal to the length of handles.
+			tblScan, ok := w.idxLookup.tblPlans[0].(*plannercore.PhysicalTableScan)
+			if ok && tblScan.NoCondition() {
```

Comment:
We don't need to check `NoCondition`, since table scan in double read can never has access conditions and filter conditions according to implementation of `convertToIndexScan`?